### PR TITLE
TF Module for secure app hosting template and add app support to ZBFW

### PR DESCRIPTION
**Pull request by rrahimm**

### CODE REVIEW

**Review by udmehrot — COMMENTED** [review]
Looks Good

**Comment by rrahimm** [comment]
Hello @danischm 

Can you review and merge.

File: sdwan_feature_templates.tf [hunk 1]
```diff
@@ -1994,3 +1994,16 @@ resource "sdwan_vpn_interface_svi_feature_template" "vpn_interface_svi_feature_t
...
+  for_each     = { for t in try(local.edge_feature_templates.secure_app_hosting_templates, {}) : t.name => t }
+  name         = each.value.name
+  description  = each.value.description
+  device_types = [for d in try(each.value.device_types, local.defaults.sdwan.edge_feature_templates.secure_app_hosting_templates.device_types) : try(local.device_type_map[d], "vedge-${d}")]
+  virtual_applications = [{
+    nat              = try(each.value.nat, null)
+    database_url     = try(each.value.download_url_database_on_device, null)
+    resource_profile = try(each.value.resource_profile, null)
+    instance_id      = 1
+  }]
```
Comment (danischm) [inline sdwan_feature_templates.tf:2008]: I think we should add the explicit "localized_policy" dependency here, as with other feature templates.

**Review by danischm — CHANGES_REQUESTED** [review]

File: sdwan_feature_templates.tf (hunk 1 above)
Comment (rrahimm) [reply sdwan_feature_templates.tf:2008]: as per my understanding of dependency, I believe we don't need it here. as the entire template is self-sufficient with direct values and no reference.

```
edge_feature_templates_secure_app_hosting:
  name: regex('^[^<>!&" ]{1,128}$')
  description: str()
  device_types: list(enum("C1121X-8PLTEP", ...... "ISR1100X-4G-XE"), min=1, required=False)
  nat: bool()
  download_url_database_on_device: bool()
  resource_profile: enum("low", "medium", "high")
```

**Comment by rrahimm** [comment]
Hello @danischm 

Can you review and approve.

**Comment by vjoshevs** [comment]
Looks ok to me. @tzarski0  do we keep update of the device list supported by SDWAN ?

**Comment by tzarski0** [comment]
The platforms you see under defaults/sdwan.yaml are just the most commonly used platforms, but this can be customized by customer if they are using different platforms.

**Comment by tzarski0** [comment]
@vjoshevs is this ready for merge? This is part of the problem we have with security policy.

**Comment by vjoshevs** [comment]
@tzarski0  yes as we spoke this is ready for merge

**Pull request merged**